### ISSUE TITLE
Very slow generation of MethodError

### ISSUE THREAD

**Issue by cafaxo** [JuliaLang/julia#38037]
I am on commit 63390c26c1.
The code
```julia
julia> [1.0, 2.0] * [[1.0], [1.0]]
```
produces a `MethodError` as expected.
However, this `MethodError` takes a very long time to generate.
First it hangs at
`ERROR: MethodError:`, then at `ERROR: MethodError: no method matching *(::Vector{Float64}, ::Vector{Vector{Float64}})`.

If I do `using Plots` (and plot some things) before executing this line, it takes even longer (30 seconds on my laptop).

Is this expected?
In version 1.5.2, the `MethodError` is generated much faster.

Edit. Times:
commit 63390c26c1:
```
time ./julia -e '[1.0, 2.0] * [[1.0], [1.0]]'
ERROR: MethodError: no method matching *(::Vector{Float64}, ::Vector{Vector{Float64}})
Closest candidates are:
  *(::Any, ::Any, ::Any, ::Any...) at operators.jl:540
  *(::StridedVecOrMat{T} where T, ::LinearAlgebra.LQPackedQ) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/LinearAlgebra/src/lq.jl:278
  *(::Union{SparseArrays.AbstractSparseMatrixCSC{TA, Ti}, SubArray{TA, 2, var"#s809", Tuple{Base.Slice{Base.OneTo{Int64}}, I}, L} where L where var"#s809"<:SparseArrays.AbstractSparseMatrixCSC{TA, Ti} where I<:AbstractUnitRange} where Ti, ::StridedVector{Tx}) where {TA, Tx} at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/SparseArrays/src/linalg.jl:47
  ...
Stacktrace:
 [1] top-level scope
   @ none:1

________________________________________________________
Executed in    6.37 secs   fish           external 
   usr time    6.24 secs  636.00 micros    6.24 secs 
   sys time    0.96 secs  295.00 micros    0.96 secs
```
1.5.2
```
time ./julia -e '[1.0, 2.0] * [[1.0], [1.0]]'
ERROR: MethodError: no method matching *(::Array{Float64,1}, ::Array{Array{Float64,1},1})
Closest candidates are:
  *(::Any, ::Any, ::Any, ::Any...) at operators.jl:538
  *(::LinearAlgebra.Transpose{var"#s828",var"#s827"} where var"#s827"<:StridedArray{T, 2} where var"#s828", ::StridedArray{S, 1}) where {T<:Union{Complex{Float32}, Complex{Float64}, Float32, Float64}, S} at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/LinearAlgebra/src/matmul.jl:84
  *(::LinearAlgebra.Adjoint{var"#s828",var"#s827"} where var"#s827"<:StridedArray{T, 2} where var"#s828", ::StridedArray{S, 1}) where {T<:Union{Complex{Float32}, Complex{Float64}, Float32, Float64}, S} at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.5/LinearAlgebra/src/matmul.jl:105
  ...
Stacktrace:
 [1] top-level scope at none:1

________________________________________________________
Executed in    2.21 secs   fish           external 
   usr time    2.13 secs  692.00 micros    2.13 secs 
   sys time    0.88 secs  127.00 micros    0.88 secs
```

**Comment by JeffBezanson** [comment]
From bisecting, this seems to have happened in maybe 2 steps, and one of them is 258d04e8bfdb8189fd61199bccd7641650fec07b. Profile:
```
ROOT                      : 14.33 %   1096159456
GC                        :  2.83 %   216215972
LOWERING                  :  0.02 %   1199052
PARSING                   :  0.00 %   337437
INFERENCE                 :  7.88 %   602702324
CODEGEN                   :  1.16 %   88598492
METHOD_LOOKUP_SLOW        :  0.04 %   2967194
METHOD_LOOKUP_FAST        :  1.83 %   140336545
LLVM_OPT                  : 29.33 %   2243581271
LLVM_MODULE_FINISH        :  0.42 %   31975680
LLVM_EMIT                 :  0.01 %   564879
METHOD_LOOKUP_COMPILE     :  0.02 %   1408902
METHOD_MATCH              : 38.81 %   2969281606
```

**Comment by JeffBezanson** [comment]
That was with the linked commit. This is the profile on master:
```
ROOT                      : 14.82 %   1152578990
GC                        :  0.63 %   48977259
LOWERING                  :  0.00 %   319061
PARSING                   :  0.00 %   338396
INFERENCE                 :  1.18 %   91972457
CODEGEN                   :  0.08 %   6559450
METHOD_LOOKUP_SLOW        :  0.02 %   1735492
METHOD_LOOKUP_FAST        :  0.27 %   20627528
LLVM_OPT                  :  2.12 %   164766957
LLVM_MODULE_FINISH        :  0.06 %   4500027
METHOD_MATCH              : 79.03 %   6145561927
```

**Comment by vtjnash** [comment]
Closed by #38061
